vllm/model_executor/layers/sampler.py

```diff
@@ -539,7 +539,11 @@ def _get_ranks(x: torch.Tensor, indices: torch.Tensor) -> torch.Tensor:
     """
     vals = x[torch.arange(0, len(x), device=x.device, dtype=indices.dtype),
              indices]
-    return (x > vals[:, None]).long().sum(1).add_(1)
+    result = (x > vals[:, None])
+    del vals
+    torch.cuda.empty_cache()
+    result = result.sum(1).add_(1)
+    return result
```

Comment:
Thanks @SwapnilDreams100! I'm not sure that we'll explicitly want to call empty_cache() here, hopefully the other changes you made should be sufficient (avoiding the long cast and deleting intermediate tensor)

```suggestion
    return result.sum(1).add_(1)
```